### PR TITLE
enabled passing of corrupt files to map

### PR DESCRIPTION
**Pull request by hannahc243**
Fixes https://github.com/sunpy/sunpy/issues/3688 which allows passing of corrupt files to map

### CODE REVIEW

**Comment by hayesla** [comment]
Thanks for the PR @hannahc243 and congrats on your first contribution ! 🥳

**Review by ayshih — COMMENTED** [review]
Thanks for the PR!  Some immediate thoughts:

- Don't add a "bad" test file to the repository, much less one that is named to look like a real file!  Such an approach is bound to cause confusion and breakages, as you've already discovered with the unrelated unit test that you had to separately fix.  Instead, define a pytest fixture that creates a temporary "bad" test file for use in the unit tests.
- Naming the keyword "silence_errors" makes me uneasy because it still emits warnings, so is definitely not silent.  I'd probably name it something like "skip_errors", but others might have better suggestions.

**Comment by ayshih** [comment]
Maybe `allow errors`?

**Comment by nabobalis** [comment]
> * Don't add a "bad" test file to the repository, much less one that is named to look like a real file!  Such an approach is bound to cause confusion and breakages, as you've already discovered with the unrelated unit test that you had to separately fix.  Instead, define a pytest fixture that creates a temporary "bad" test file for use in the unit tests.

I don't see the harm in a bad test file, I was the one who told them to add it, the filename I think we can work on. It would be nice to have a file that is from a failed download, that we can use easily. 

That unrelated unit test I don't think even needs to exist. 

> * Naming the keyword `silence_errors` makes me uneasy because it still emits warnings, so is definitely not silent.  I'd probably name the keyword something like `skip_errors`, but others might have better suggestions.

That `silence_errors` already exists in the Map API, I wanted to avoid adding a new keyword. If you prefer, we can add a new keyword that will handle this behaviour, but since "silence_errors" already does this but for hdu pairs that are passed, extending it to do it for files seems fine to me.

**Comment by nabobalis** [comment]
> Well, then I additionally prefer to change the name of the keyword that already exists in the API.  It doesn't make sense that a keyword called `silence_errors` does not actually silence anything, but instead replaces errors with (non-silent) warnings.  

I was hoping you wouldn't say this but I agree.

File: changelog/7018.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+We added the ability to pass corrupt fits files to map, using the existing keyword argument ``silence_errors``.
```
Comment (alasdairwilson) [inline changelog/7018.feature.rst:1]: Maybe something like this for the changelog?:

```suggestion
Added the ability to skip over errors raised when passing invalid fits files to map using the existing keyword argument ``silence_errors``.
```

File: sunpy/data/test/aia_lev1_211a_2022_03_30t17_21_21_63z_image_lev1.fits [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
```
Comment (nabobalis) [inline sunpy/data/test/aia_lev1_211a_2022_03_30t17_21_21_63z_image_lev1.fits:1]: We should rename this file to something else: `not_actually_a_fits.fits`

File: sunpy/io/tests/test_fits.py [hunk 3]
```diff
@@ -138,7 +138,8 @@ def test_fitsheader():
     """Test that all test data can be converted back to a FITS header."""
     extensions = ('.fts', '.fits')
     for ext in extensions:
-        test_files = [f for f in get_test_data_filenames() if f.suffix == ext]
+        bad_file = "aia_lev1_211a_2022_03_30t17_21_21_63z_image_lev1.fits"
```
Comment (nabobalis) [inline sunpy/io/tests/test_fits.py:141]:
```suggestion
        bad_file = "not_actually_fits.fits"
```
When the file has been renamed.

File: sunpy/map/tests/test_map_factory.py [hunk 4]
```diff
@@ -315,3 +315,21 @@ def test_map_fits():
...
     assert isinstance(fits_map, sunpy.map.GenericMap)
     assert fits_map.data.base is not None
+
+def test_map_list_of_files():
+    files = [AIA_171_IMAGE, get_test_filepath('aia_lev1_211a_2022_03_30t17_21_21_63z_image_lev1.fits')]
+    with pytest.warns(SunpyUserWarning, match='Failed to read'):
+        amap = sunpy.map.Map(files, silence_errors=True)
+        assert amap.data.shape == (128,128)
+
+    files = [AIA_171_IMAGE, get_test_filepath('aia_lev1_211a_2022_03_30t17_21_21_63z_image_lev1.fits'), AIA_171_IMAGE]
```
Comment (nabobalis) [inline sunpy/map/tests/test_map_factory.py:325]:
```suggestion
    files = [AIA_171_IMAGE, get_test_filepath('not_actually_fits.fits'), AIA_171_IMAGE]
```
When the file has been renamed.

File: changelog/7018.feature.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Added the ability to skip over errors raised when passing invalid fits files to map using the existing keyword argument ``silence_errors``.
```
Comment (hayesla) [inline changelog/7018.feature.rst:1]:
```suggestion
Added the ability to skip over errors raised for invalid fits files when passing a list of files to map using the existing keyword argument ``silence_errors``.
```

**Review by hayesla — APPROVED** [review]
Thanks for this PR @hannahc243 !

Looks great, just two suggestions

File: sunpy/map/map_factory.py [hunk 6]
```diff
@@ -97,7 +97,11 @@ def _read_file(self, fname, **kwargs):
         try:
             pairs = read_file(os.fspath(fname), **kwargs)
         except Exception as e:
-            raise OSError(f"Failed to read {fname}\n{e}") from e
+            msg = f"Failed to read {fname}\n{e}"
+            if kwargs.get("silence_errors"):
+                warn_user(msg)
+                return []
+            raise OSError(msg) from e
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/map/map_factory.py:104]: I think this is worth adding to my follow up PR renaming the keyword.

**Review by nabobalis — APPROVED** [review]

**Review by alasdairwilson — APPROVED** [review]

**Comment by nabobalis** [comment]
Thank you for the pull request @hannahc243!

**Pull request merged**